warehouse/redshift_test.go

```diff
@@ -114,3 +122,137 @@ func TestGetBucketAndKey(t *testing.T) {
 		}
 	}
 }
+
+func TestValidateSchemaConfig(t *testing.T) {
+
+	testCases := []struct {
+		conf       *config.Config
+		hasError   bool
+		errMessage string
+	}{
+		{
+			conf:       makeConf(""),
+			hasError:   true,
+			errMessage: "DatabaseSchema definition missing from Redshift configuration. More information: https://github.com/fullstorydev/hauser/blob/master/Redshift.md#details-about-database-schema-configuration",
+		},
+		{
+			conf:       makeConf("test"),
+			hasError:   false,
+			errMessage: "",
+		},
+		{
+			conf:       makeConf("search_path"),
+			hasError:   false,
+			errMessage: "",
+		},
+	}
+
+	for _, tc := range testCases {
+		wh := NewRedshift(tc.conf)
+		err := wh.validateSchemaConfig()
+		if tc.hasError && err == nil {
+			t.Errorf("expected Redshift.validateSchemaConfig() to return an error when config.Config.Redshift.DatabaseSchema is empty")
+		}
+		if tc.hasError && err.Error() != tc.errMessage {
```

Comment:
I think checking for an exact error message makes this a fragile test, and doesn't add much value at the moment since the function only returns one type of error. We're likely to change the error message thinking it has no consequences. Although fixing a unit test is a small consequence, it's a bit of extra work we can avoid.

Reply:
This thought occurred to me as well - I appreciate the general brittleness that comes with testing error message semantics. Having said that: I'd like to keep this one in place since this message is a crucial element to communicating new behavior to existing hauser users (when they decide to upgrade to the latest version). My expectation is that existing users will get the latest version and might very well forget to add the `DatabaseSchema` parameter to their config. They should immediately be informed that hauser won't work this way (and why) and what they can do to fix it.

Reply:
I agree that the error message is important, but still disagree that its exact content needs testing. This is already merged, but I feel pretty strongly on this subject, and we can maybe hash it out f2f if this comes up in future PRs.

Reply:
Hi Jess,

Understood - I've got another PR coming to verify the invocation of ValidateDatabaseSchema + refactor validation into config. The branch is out, I just haven't made the PR yet (I wanted to stew on it until Wednesday). We can discuss the error condition test and make a change as part of that PR if this works for you.

Thank you,

-Patrick